packages/framework/src/parser.ts

```diff
@@ -197,17 +199,31 @@ export const unwrapWidget = (obj: Record<string, unknown>): EnsembleWidget => {
     const unwrappedTemplate = unwrapWidget(template as Record<string, unknown>);
     set(properties as object, ["item-template", "template"], unwrappedTemplate);
   }
-  if (isArray(items) && !isEmpty(items)) {
-    if ("widget" in items[0]) {
-      const valueItems = (items as Record<string, unknown>[]).map(
-        ({ label, widget, icon }) => {
-          const unwrappedWidget = unwrapWidget(
-            widget as Record<string, unknown>,
-          );
-          return { label, icon, widget: unwrappedWidget };
-        },
-      );
-      set(properties as object, "items", valueItems);
+  if (!isEmpty(items) && isArray(items)) {
+    if (isString(items[0])) {
+      set(properties as object, "items", items);
```

Comment:
We can always set the items property or leave it unchanged, and then just handle the special structure in your widget. Otherwise this will get very complicated trying to handle all the cases in the parser.